tla/consensus/Network.tla

```diff
@@ -98,7 +98,7 @@ LOCAL OrderOneMoreMessage(m) ==
     \/ Len(SelectSeq(messages[m.dest], LAMBDA e: m = e)) < Len(SelectSeq(messages'[m.dest], LAMBDA e: m = e))
 
 LOCAL OrderDropMessages(server) ==
-    \E s \in Suffixes(messages[server]):  \* TODO - Change to SubSeqs if more sophisticated message loss is needed.
+    \E s \in AllSubSeqs(messages[server]):
```

Comment:
@lemmy Looks like this causes a large increase in trace validation time, in fact so large that it's causing a timeout in the CI with `suffix_collision.2` still running after an hour.

Comparing the validation time for `suffix_collision.1` with the different options here.

### Suffixes
```
Finished computing initial states: 1 distinct state generated at 2024-01-23 10:06:43.
Progress(187) at 2024-01-23 10:06:44: 441 states generated, 421 distinct states found, 0 states left on queue.
Checking temporal properties for the complete state space with 421 total distinct states at (2024-01-23 10:06:44)
Finished checking temporal properties in 00s at 2024-01-23 10:06:44
Model checking completed. No error has been found.
  Estimates of the probability that TLC did not check all reachable states
  because two distinct states had the same fingerprint:
  calculated (optimistic):  val = 4.6E-16
441 states generated, 421 distinct states found, 0 states left on queue.
The depth of the complete state graph search is 187.
The average outdegree of the complete state graph is 1 (minimum is 0, the maximum 4 and the 95th percentile is 1).
Finished in 02s at (2024-01-23 10:06:44)
```

### SubSeqs
```
Finished computing initial states: 1 distinct state generated at 2024-01-23 10:08:14.
Progress(165) at 2024-01-23 10:08:17: 511 states generated (511 s/min), 450 distinct states found (450 ds/min), 23 states left on queue.
Progress(187) at 2024-01-23 10:08:17: 781 states generated, 717 distinct states found, 0 states left on queue.
Checking temporal properties for the complete state space with 717 total distinct states at (2024-01-23 10:08:17)
Finished checking temporal properties in 00s at 2024-01-23 10:08:17
Model checking completed. No error has been found.
  Estimates of the probability that TLC did not check all reachable states
  because two distinct states had the same fingerprint:
  calculated (optimistic):  val = 2.5E-15
781 states generated, 717 distinct states found, 0 states left on queue.
The depth of the complete state graph search is 187.
The average outdegree of the complete state graph is 1 (minimum is 0, the maximum 8 and the 95th percentile is 1).
Finished in 05s at (2024-01-23 10:08:17)
```

### AllSubSeqs
```
Finished computing initial states: 1 distinct state generated at 2024-01-23 10:08:58.
Progress(161) at 2024-01-23 10:09:03: 285 states generated (285 s/min), 285 distinct states found (285 ds/min), 18 states left on queue.
Progress(162) at 2024-01-23 10:10:06: 346 states generated (61 s/min), 346 distinct states found (61 ds/min), 63 states left on queue.
Progress(187) at 2024-01-23 10:10:07: 885 states generated, 793 distinct states found, 0 states left on queue.
Checking temporal properties for the complete state space with 793 total distinct states at (2024-01-23 10:10:07)
Finished checking temporal properties in 00s at 2024-01-23 10:10:07
Model checking completed. No error has been found.
  Estimates of the probability that TLC did not check all reachable states
  because two distinct states had the same fingerprint:
  calculated (optimistic):  val = 4.0E-15
885 states generated, 793 distinct states found, 0 states left on queue.
The depth of the complete state graph search is 187.
The average outdegree of the complete state graph is 1 (minimum is 0, the maximum 8 and the 95th percentile is 1).
Finished in 01min 11s at (2024-01-23 10:10:07)
```

Going from `Suffixes` to `AllSubSeqs` increased run time from 2s to 1m11s, despite only doubling the number of generated states. Is `AllSubSeqs` slow to evaluate, or hasssss this generated far more candidate states under the hood that aren't reported?

Reply:
I will look into it.  `Suffixes` has a [Java module override](https://github.com/tlaplus/CommunityModules/blob/e6a395429682a90f6b2f36541d08faa34ef3394c/modules/tlc2/overrides/SequencesExt.java#L574-L607), but that might not be it.

Reply:
Let's cross that bridge once we get there, but we might have to finally bite the bullet and implement a trace checking mode in TLC, as discussed in https://github.com/microsoft/CCF/blob/2d66dfe6d0523de3d705d49a8f0bc26a29c73db5/tla/consensus/Traceccfraft.tla#L379-L383.

That mode would also address https://github.com/microsoft/CCF/pull/5933